association.go

```diff
@@ -277,45 +302,56 @@ func (a *Association) sendCookieEcho() error {
 
 // Close ends the SCTP Association and cleans up any state
 func (a *Association) Close() error {
+	a.log.Debugf("[%s] closing association..", a.name())
+
+	// TODO: shutdown sequence
+	a.lock.Lock()
+	a.setState(closed)
+	a.lock.Unlock()
+
 	err := a.netConn.Close()
 	if err != nil {
 		return err
 	}
 
-	// Stop all retransmission timers
-	a.t1Init.stop()
-	a.t1Cookie.stop()
-	a.t3RTX.stop()
+	// Close all retransmission timers
+	a.t1Init.close()
+	a.t1Cookie.close()
+	a.t3RTX.close()
 
 	// Wait for readLoop to end
 	<-a.closeCh
 
+	a.log.Debugf("[%s] association closed", a.name())
 	return nil
 }
 
 func (a *Association) readLoop() {
+	var closeErr error
 	defer func() {
 		a.lock.Lock()
-		closeErr := errors.New("association closed")
+		//closeErr := errors.New("association closed")
```

Comment:
Should we delete this line (or uncomment)?

Reply:
will delete it.

Reply:
Fixed at ef7368e